### ISSUE TITLE
CloudWatch Logs: PutLogEvents cause Error with multi-byte message

### ISSUE THREAD

**Issue by fujitalon** [localstack/localstack#1828]
Hi, there.

I try to putLog to CloudWatch logs with localstack. When I use multi-byte character message(in Japanese, UTF-8), I got error from localstack.

There is no error with ascii only message.
No error with aws CloudWatch logs.

Received error:
```
api_1         | [worker-1] cwLogs putLogEvent failed.
api_1         | SerializationError: failed to unmarshal error message
api_1         |         status code: 500, request id:
api_1         | caused by: UnmarshalError: failed decoding error message
api_1         |         00000000  3c 21 44 4f 43 54 59 50  45 20 48 54 4d 4c 20 50  |<!DOCTYPE HTML P|
api_1         | 00000010  55 42 4c 49 43 20 22 2d  2f 2f 57 33 43 2f 2f 44  |UBLIC "-//W3C//D|
api_1         | 00000020  54 44 20 48 54 4d 4c 20  33 2e 32 20 46 69 6e 61  |TD HTML 3.2 Fina|
api_1         | 00000030  6c 2f 2f 45 4e 22 3e 0a  3c 74 69 74 6c 65 3e 35  |l//EN">.<title>5|
api_1         | 00000040  30 30 20 49 6e 74 65 72  6e 61 6c 20 53 65 72 76  |00 Internal Serv|
api_1         | 00000050  65 72 20 45 72 72 6f 72  3c 2f 74 69 74 6c 65 3e  |er Error</title>|
api_1         | 00000060  0a 3c 68 31 3e 49 6e 74  65 72 6e 61 6c 20 53 65  |.<h1>Internal Se|
api_1         | 00000070  72 76 65 72 20 45 72 72  6f 72 3c 2f 68 31 3e 0a  |rver Error</h1>.|
api_1         | 00000080  3c 70 3e 54 68 65 20 73  65 72 76 65 72 20 65 6e  |<p>The server en|
api_1         | 00000090  63 6f 75 6e 74 65 72 65  64 20 61 6e 20 69 6e 74  |countered an int|
api_1         | 000000a0  65 72 6e 61 6c 20 65 72  72 6f 72 20 61 6e 64 20  |ernal error and |
api_1         | 000000b0  77 61 73 20 75 6e 61 62  6c 65 20 74 6f 20 63 6f  |was unable to co|
api_1         | 000000c0  6d 70 6c 65 74 65 20 79  6f 75 72 20 72 65 71 75  |mplete your requ|
api_1         | 000000d0  65 73 74 2e 20 45 69 74  68 65 72 20 74 68 65 20  |est. Either the |
api_1         | 000000e0  73 65 72 76 65 72 20 69  73 20 6f 76 65 72 6c 6f  |server is overlo|
api_1         | 000000f0  61 64 65 64 20 6f 72 20  74 68 65 72 65 20 69 73  |aded or there is|
api_1         | 00000100  20 61 6e 20 65 72 72 6f  72 20 69 6e 20 74 68 65  | an error in the|
api_1         | 00000110  20 61 70 70 6c 69 63 61  74 69 6f 6e 2e 3c 2f 70  | application.</p|
api_1         | 00000120  3e 0a                                             |>.|
api_1         |
api_1         | caused by: invalid character '<' looking for beginning of value
```

localstack's output:
```
localstack_1  | 172.18.0.6 - - [05/Dec/2019 08:13:48] "POST / HTTP/1.1" 500 -
localstack_1  | Error on request:
localstack_1  | Traceback (most recent call last):
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/werkzeug/serving.py", line 302, in run_wsgi
localstack_1  | execute(self.server.app)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/werkzeug/serving.py", line 290, in execute
localstack_1  | application_iter = app(environ, start_response)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/moto/server.py", line 104, in __call__
localstack_1  | return backend_app(environ, start_response)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
localstack_1  | return self.wsgi_app(environ, start_response)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
localstack_1  | response = self.handle_exception(e)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
localstack_1  | reraise(exc_type, exc_value, tb)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
localstack_1  | raise value
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
localstack_1  | response = self.full_dispatch_request()
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
localstack_1  | rv = self.handle_user_exception(e)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
localstack_1  | reraise(exc_type, exc_value, tb)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
localstack_1  | raise value
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
localstack_1  | rv = self.dispatch_request()
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
localstack_1  | return self.view_functions[rule.endpoint](**req.view_args)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/moto/core/utils.py", line 140, in __call__
localstack_1  | result = self.callback(request, request.url, {})
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/moto/core/responses.py", line 117, in dispatch
localstack_1  | return cls()._dispatch(*args, **kwargs)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/moto/core/responses.py", line 199, in _dispatch
localstack_1  | self.setup_class(request, full_url, headers)
localstack_1  | File "/opt/code/localstack/.venv/lib/python3.6/site-packages/moto/core/responses.py", line 155, in setup_class
localstack_1  | querystring.update(parse_qs(raw_body, keep_blank_values=True))
localstack_1  | File "/usr/lib/python3.6/urllib/parse.py", line 656, in parse_qs
localstack_1  | max_num_fields=max_num_fields)
localstack_1  | File "/usr/lib/python3.6/urllib/parse.py", line 691, in parse_qsl
localstack_1  | qs, _coerce_result = _coerce_args(qs)
localstack_1  | File "/usr/lib/python3.6/urllib/parse.py", line 123, in _coerce_args
localstack_1  | return _decode_args(args) + (_encode_result,)
localstack_1  | File "/usr/lib/python3.6/urllib/parse.py", line 107, in _decode_args
localstack_1  | return tuple(x.decode(encoding, errors) if x else '' for x in args)
localstack_1  | File "/usr/lib/python3.6/urllib/parse.py", line 107, in <genexpr>
localstack_1  | return tuple(x.decode(encoding, errors) if x else '' for x in args)
localstack_1  | UnicodeDecodeError: 'ascii' codec can't decode byte 0xe7 in position 287: ordinal not in range(128)
```

**Comment by whummer** [comment]
Thanks for reporting @fujitalon . I haven't been able to reproduce this issue - seems to be working in the latest version. Added a test here to avoid regressions: #1842

From your log output it seems that you're using an older version of the Docker image - `/usr/lib/python3.6` does not exist in the latest image, it is now `/usr/lib/python3.7`. Can you please do a `docker pull localstack/localstack` and give it another try?

Closing this issue - please re-open if the problem persists.

**Comment by fujitalon** [comment]
@whummer 
I retry latest version of localstack, but not  fixed.
So I made error reproduction code at:
https://github.com/fujitalon/localstack-cwlogs-error

Clone this repository and run 'docker-compose up', error displayed in console log.

**Comment by whummer** [comment]
Thanks for providing the sample project @Arthamys . This should be fixed in #1863 . Can you please give it another try with the latest version of the Docker image? Please report here if the problem persists. Thanks
